File: src/main/java/ua/softgroup/matrix/server/persistent/repository/UserRepository.java
package ua.softgroup.matrix.server.persistent.repository;

import org.springframework.data.repository.CrudRepository;
import ua.softgroup.matrix.server.persistent.entity.User;

public interface UserRepository extends CrudRepository<User, Long> {

    /**
     * Returns a {@code User} by the given {@code username}.
     *
     * @param username userName of the requested user
     * @return a {@code User} or {@code null} if a {@code User} with the given {@code username} not found
     */
    User findByUsername(String username);

}

File: src/main/scala/ua/softgroup/matrix/server/persistent/repository/UserRepository.scala
package ua.softgroup.matrix.server.persistent.repository

import org.springframework.data.repository.CrudRepository
import ua.softgroup.matrix.server.persistent.entity.User

/**
  * @author Oleksandr Tyshkovets <sg.olexander@gmail.com> 
  */
trait UserRepository extends CrudRepository[User, java.lang.Long] {

  /**
    * Returns a ''User'' by the given ''username''
    *
    * @param username username of the requested user
    * @return ''User'' or ''null'' if a user with the given ''username'' not found
    */
  def findByUsername(username: String): User

  def findByTrackerToken(token: String): User

}
